### PR TITLE
[PATCH v3] test: packet_gen: add VLAN option

### PR DESCRIPTION
**Pull request by psavol**
Added -v option to insert VLAN(s) into the packets. VLAN header
TPID and TCI fields (<tpid>:<tci>) are defined in hexadecimal.
Multiple VLANs may be stacked with a comma separated list,
starting from the outter most VLAN.

Signed-off-by: Petri Savolainen <petri.savolainen@nokia.com>

### CODE REVIEW

File: test/performance/odp_packet_gen.c [hunk 1]
```diff
@@ -114,6 +121,10 @@ static void print_usage(void)
 	       "                          addresses (no spaces), one address per packet IO\n"
 	       "                          interface e.g. AA:BB:CC:DD:EE:FF,11:22:33:44:55:66\n"
 	       "                          Default per interface: 02:00:00:A0:B0:CX, where X = 0,1,...\n"
+	       "  -v, --vlan <tpid:tci>   VLAN configuration. Comma-separated list of VLAN TPID:TCI\n"
+	       "                          values in hexadecimal. For example:\n"
+	       "                          VLAN 200 (decimal):          8100:c8\n"
+	       "                          Double tagged VLANs 1 and 2: 88a8:1,8100:2\n"
```
Comment (matiaselo) [inline test/performance/odp_packet_gen.c]: You could mention the order of the VLAN headers here.

File: test/performance/odp_packet_gen.c [hunk 2]
```diff
@@ -451,6 +512,10 @@ static int open_pktios(test_global_t *global)
 	printf("  tx burst gap        %" PRIu64 " nsec\n",
 	       test_options->gap_nsec);
 	printf("  clock resolution    %" PRIu64 " Hz\n", odp_time_local_res());
+	for (i = 0; i < test_options->num_vlan; i++) {
```
Comment (matiaselo) [inline test/performance/odp_packet_gen.c]: VLAN headers are missing from the following pool `seg_len` calculation:
`	seg_len = ODPH_ETHHDR_LEN + ODPH_IPV4HDR_LEN + ODPH_UDPHDR_LEN;`

File: test/performance/odp_packet_gen.c (hunk 1 above)
Comment (psavol) [reply test/performance/odp_packet_gen.c]: fixed in v2

File: test/performance/odp_packet_gen.c (hunk 2 above)
Comment (psavol) [reply test/performance/odp_packet_gen.c]: fixed in v2

File: test/performance/odp_packet_gen.c [hunk 3]
```diff
@@ -114,6 +122,11 @@ static void print_usage(void)
 	       "                          addresses (no spaces), one address per packet IO\n"
 	       "                          interface e.g. AA:BB:CC:DD:EE:FF,11:22:33:44:55:66\n"
 	       "                          Default per interface: 02:00:00:A0:B0:CX, where X = 0,1,...\n"
+	       "  -v, --vlan <tpid:tci>   VLAN configuration. Comma-separated list of VLAN TPID:TCI\n"
+	       "                          values in hexadecimal, starting from the outter most VLAN.\n"
```
Comment (matiaselo) [inline test/performance/odp_packet_gen.c]: Typo: outter -> outer

**Pull request merged**